### PR TITLE
Locked groups: auto lock setting needs to suggest valid entries

### LINKED ISSUE #131587: Locked groups: auto lock setting needs to suggest valid entries
Testing #131196

How do I know what to put in this setting? I have no idea what the ids should be from the setting.

### PR DESCRIPTION
**Pull request by bpasero**
This PR fixes #131587

### CODE REVIEW

File: src/vs/workbench/contrib/preferences/browser/settingsWidgets.ts [hunk 1]
```diff
@@ -1372,7 +1372,7 @@ export class ObjectSettingCheckboxWidget extends AbstractListSettingWidget<IObje
 		rowElement.appendChild(element);
 
 		const valueElement = DOM.append(rowElement, $('.setting-list-object-value'));
-		valueElement.textContent = changedItem.key.data;
+		valueElement.textContent = item.keyDescription ? `${item.keyDescription} (${item.key.data})` : item.key.data;
```
Comment (bpasero) [inline src/vs/workbench/contrib/preferences/browser/settingsWidgets.ts:1375]: @rzhao271 fyi this allows me to show a nice label next to the checkbox:

image

File: src/vs/workbench/services/editor/browser/editorResolverService.ts [hunk 2]
```diff
@@ -42,6 +43,10 @@ type RegisteredEditors = Array<RegisteredEditor>;
 export class EditorResolverService extends Disposable implements IEditorResolverService {
 	readonly _serviceBrand: undefined;
 
+	// Events
+	private readonly _onDidChangeEditorRegistrations = this._register(new Emitter<void>());
```
Comment (bpasero) [inline src/vs/workbench/services/editor/browser/editorResolverService.ts:47]: @lramos15 fyi I added this event to know when registrations come and go (also added a test)

File: src/vs/workbench/services/editor/browser/editorResolverService.ts [hunk 3]
```diff
@@ -333,12 +342,19 @@ export class EditorResolverService extends Disposable implements IEditorResolver
 		});
 	}
 
-	public getEditorIds(resource: URI): string[] {
-		const editors = this.findMatchingEditors(resource);
-		if (editors.find(e => e.editorInfo.priority === RegisteredEditorPriority.exclusive)) {
-			return [];
+	public getEditors(resource?: URI): RegisteredEditorInfo[] {
```
Comment (bpasero) [inline src/vs/workbench/services/editor/browser/editorResolverService.ts:345]: @lramos15 fyi I changed this method to allow me to return all registered editors.

Comment (lramos15) [reply src/vs/workbench/services/editor/browser/editorResolverService.ts:345]: Is there a reason we now have `getEditors` and `getEditorIds`?

**Pull request merged**